doc/design/audit-trail-mvp.md

```diff
@@ -0,0 +1,157 @@
+collaboration.
+
+## Requirements
+
+1. **Persistent tracking of significant actions.** Whenever a *significant
+   action* (see list [below](#significant-actions)) occurs successfully, the archive will generate a
+   persistent record of that action, including the following data:
+    - The date and time
+    - The authorized user who triggered the action (note that this may be an actual user or the system itself for internal automated actions)
+    - The ID of the Dandiset on which the action occurs
+    - Event-specific data about the action itself
+
+2. **Display of audit log (to owners and admins only).** The Dandiset landing
+   page will display a rendered (i.e. human-readable) version of the log event
+   data (as long as the viewing user is an owner of the Dandiset, or an admin).
+
+   A possible example of such a display follows (the final rendered form may
+   differ from this depending on best practices, etc.):
+    
+    ```
+    YYYYMMDDTHHMMSS.mmmm: X added asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X updated asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X removed asset at path (/full/path) checksum
+    YYYYMMDDTHHMMSS.mmmm: X updated dandiset metadata
+    ```
+    
+3. **Extensible significant action list.** The significant actions the system is
+   capable of tracking should be extensible--that is, in the future new actions
+   should be able to be added to the system, and existing actions should be able
+   to expand the specific data they carry.
+
+## Significant Actions
+
+The following is the list of actions performed on Dandisets that are considered
+"significant" (i.e., require an audit record to be generated), along with any
+additional, event-specific data each will carry with it.
+
+| Action                  | General data                    | Specific data                                                                   |
+| ---                     | ---                             | ---                                                                             |
+| Create Dandiset         | date_time, user_id, dandiset_id | metadata, embargoed                                                             |
+| Add owner               | date_time, user_id, dandiset_id | owner_id                                                                        |
+| Remove owner            | date_time, user_id, dandiset_id | owner_id                                                                        |
+| Update Version metadata | date_time, user_id, dandiset_id | metadata                                                                        |
+| Add asset               | date_time, user_id, dandiset_id | asset_path, checksum, metadata, blob_id, asset_id                               |
+| Update asset            | date_time, user_id, dandiset_id | asset_path, old_asset_path, checksum, blob_id, metadata, asset_id, old_asset_id |
```

Comment:
I thought I left the comment before but I fail to find resolved conversation, sorry if repeating but seems this does not acknowledge the fact that we have now a multitude of blob vs zarr. So either we need 
```suggestion
| Add asset               | date_time, user_id, dandiset_id | asset_path, checksum, metadata, blob_id, zarr_id, asset_id                               |
| Update asset            | date_time, user_id, dandiset_id | asset_path, old_asset_path, checksum, blob_id, zarr_id, metadata, asset_id, old_asset_id |
```

or (more flexible/scalable for the future possible addition of some other beastie)

```suggestion
| Add asset               | date_time, user_id, dandiset_id | asset_path, checksum, metadata, object_id, object_type, asset_id                               |
| Update asset            | date_time, user_id, dandiset_id | asset_path, old_asset_path, checksum, object_id, object_type, metadata, asset_id, old_asset_id |
```

where `object_type` could be `blob` or `zarr` (not sure if we have semantic already for this)